v03_pipeline/lib/misc/io.py

```diff
@@ -144,18 +169,17 @@ def write(
     t: hl.Table | hl.MatrixTable,
     destination_path: str,
     checkpoint: bool = True,
-    n_partitions: int | None = None,
 ) -> hl.Table | hl.MatrixTable:
     suffix = 'mt' if isinstance(t, hl.MatrixTable) else 'ht'
+    read_fn = hl.read_matrix_table if isinstance(t, hl.MatrixTable) else hl.read_table
     if checkpoint:
-        t = t.checkpoint(
-            os.path.join(
-                Env.HAIL_TMPDIR,
-                f'{uuid.uuid4()}.{suffix}',
-            ),
+        path = os.path.join(
+            Env.HAIL_TMPDIR,
+            f'{uuid.uuid4()}.{suffix}',
         )
-    # "naive_coalesce" will decrease parallelism of hail's pipelined operations
-    # , so we sneak this re-partitioning until after the checkpoint.
-    if n_partitions:
+        # not using checkpoint to read/write here because the checkpoint codec is different, leading to a different on disk size.
+        t.write(path)
```

Comment:
I feel like writing twice is not ideal but I can't think of a better way to do it, unless we wanted to do some sort of really rough estimate involving the number of rows. Is there a reason that this is nested inside an `if checkpoint:` block though? I feel like we would always want to compute the correct partition number.

Reply:
I think there isn't a good reason at this point, so I will change it!  Originally, I believe the family tables were not checkpointed because they are freshly written each time, but that isn't happening any more.  Pretty sure `checkpoint` is never false at this point, in looking at the function calls in the rest of the codebase, so I'll just remove it.